src/lib.rs

```diff
@@ -53,62 +53,43 @@
 //! assert_eq!(lzjd, 0.5714285714285714);
 //! ```
 
-#[macro_use]
-extern crate failure_derive;
-
 pub use crate::lz_dict::LZDict;
 pub use murmurhash3::Murmur3HashState;
-use std::io;
 
 /// crc32 wrapper;
 pub mod crc32;
 /// LZ dictionary implementation
 pub mod lz_dict;
 
-#[derive(Debug, Fail)]
+#[derive(Debug)]
 pub enum LZJDError {
-    #[fail(display = "IO error: {}", err)]
-    Io {
-        #[cause]
-        err: io::Error,
-    },
-    #[fail(display = "Decode error: {}", err)]
-    Base64 {
-        #[cause]
-        err: base64::DecodeError,
-    },
-    #[fail(display = "Bincode error: {}", err)]
-    Bincode {
-        #[cause]
-        err: bincode::Error,
-    },
-    #[fail(display = "Error: {}", msg)]
-    Msg { msg: String },
+    Io(String),
```

Comment:
This drops the error information, which might not be ideal.

Reply:
I'm not sure of the best way to store the error information, I think just one string would be fine. I don't know that various enum types are helpful. But that crate had to be removed due to it also be abandoned and having security problems.